src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -220,6 +223,21 @@ protected void AddMapInternal(string collection, string subCollecction, Indexing
             funcs.Add(map);
         }
 
+        internal void CheckDepthOfStackInOutputMap(IndexDefinition indexMetadata, DocumentDatabase documentDatabase)
+        {
+            var performanceHintConfig = documentDatabase.Configuration.PerformanceHints;
+            if (StackSizeInSelectClause > performanceHintConfig.MaxDepthOfRecursionInLinqSelect)
+            {
+                documentDatabase.NotificationCenter.Add(PerformanceHint.Create(
+                    documentDatabase.Name,
+                    $"Index '{indexMetadata.Name}' contains a lot of `let` clause.",
+                    $"Each of let clause is nesting projections and could potentially leads to StackoverflowException.",
```

Comment:
```suggestion
                    $"Each of the let clause is nesting projections and could potentially leads to StackoverflowException.",
```

Reply:
Need to also include a proper fix for that. Include a sample of a bad index and a transformation for a good one

Reply:
Including a sample can be a bit problematic due to lack of space and proper formatting for the notifications in the studio. I think we already have a section in the docs where we explain the indexing alerts.